frontend/src/hooks/useModalContext.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from 'react';
+import { useContext } from 'react';
+
+import { ModalProps } from './../components/common/GenericModal';
+import { ModalContext } from './../contexts/modalContext';
+export const useModalContext = (newModalProps?: ModalProps) => {
+  const { modalProps, setModalProps, setDisplayModal } = useContext(ModalContext);
+  useEffect(() => {
+    if (newModalProps !== undefined) {
+      setModalProps({ ...newModalProps });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Still learning react, so this might be dumb. Is the newModalProps and setModalProps not part of the dependencies of this useEffect?

Reply:
yes, maybe I need a comment here. The intent here is to force the setModalProps logic to only run once during component initialization.